pkgs/build-support/rust/fetch-cargo-tarball/default.nix

```diff
@@ -30,7 +30,9 @@ in
 let hash_ =
   if args ? hash then { outputHashAlgo = null; outputHash = args.hash; }
   else if args ? sha256 then { outputHashAlgo = "sha256"; outputHash = args.sha256; }
-  else throw "fetchCargoTarball requires a hash for ${name}";
+  # Prints 'warning: empty hash found, assuming <lib.fakeHash>', generates the FOD hash, and exits afterwards.
+  # This is useful because it allows you to generate the output hash without any boilerplate.
```

Comment:
```suggestion
```

Technically we're not supposed to do this.  Everybody does though, but we probably shouldn't *encourage* it.

https://github.com/NixOS/nixpkgs/blob/ddb18ca71d12ea4edfebe7a9241419ab6fd4f2e7/doc/contributing/coding-conventions.chapter.md?plain=1#L503

IMHO we should instead add some kind of `require-tls` option, but that probably has to be threaded down through nixpkgs and into nix itself.

Reply:
Also please update the manual:

https://github.com/NixOS/nixpkgs/blob/ddb18ca71d12ea4edfebe7a9241419ab6fd4f2e7/doc/languages-frameworks/rust.section.md?plain=1#L64

The linked paragraph, the paragraph after it, and the sentence before it can be deleted.  The sentence before *that* might require a wording change.

Reply:
I've updated the manual, let me know what you think.